### PR TITLE
The best quiz screen youll ever lay your eyes on

### PR DESCRIPTION
**Pull request by mike10911**
Works with mock data. My Mongo is acting up so I just commented out all the calls. Once Justin deploys his cluster I can use that :)

### CODE REVIEW

**Comment by alyssamui** [comment]
nit: would you be able to update the styling a bit to look more like canvas
- buttons to match w the `<hr/>` under them
image
- align the text to the left instead of the center here
image

**Comment by alyssamui** [comment]
lastly some functionality that's required in the checklist and not in this PR, can you add:
image

File: src/Kanbas/Courses/Quizzes/Details/index.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                    <td><b>Available From</b></td>
+                    <td><b>Until</b></td>
+                </tr>
+                <tr>
+                    <td className='date'>{new Date(quiz.dueDate).toLocaleString()}</td>
+                    <td className='date'>{new Date(quiz.availableDate).toLocaleString()}</td>
+                    <td className='date'>{new Date(quiz.untilDate).toLocaleString()}</td>
+                </tr>
+            </table>
+            {/* Render quiz questions and options here */}
```
Comment (jpwaves) [inline src/Kanbas/Courses/Quizzes/Details/index.tsx]: is this something we still need to do or is this covered by someone else's work?

File: src/Kanbas/Courses/Quizzes/QuizList.tsx [hunk 2]
```diff
@@ -1,5 +1,197 @@
...
+const QUIZ_API = `${API_BASE}/api`;
+
+
+interface Question {
+    _id: string;
+    type: string;
+    points: number;
+    description: string;
+    options: (string | number)[];
+    answers: (string | number)[];
```
Comment (jpwaves) [inline src/Kanbas/Courses/Quizzes/QuizList.tsx]: i might be wrong/misunderstanding how this interface is being, but i think these are both stored as `string[]` in our db and not `(string | number)[]`

File: src/Kanbas/Courses/Quizzes/QuizList.tsx [hunk 3]
```diff
@@ -1,5 +1,197 @@
...
+                                }`}
+                            >
+                                {quiz.isPublished ?  <FaCheckCircle className='text-success' /> : "🚫"}
+                            </span>
+                            <div
+                                onClick={(e) => e.stopPropagation()}
+                                className="quiz-context-menu"
+                            >
+                                {/* Implement the context menu here */}
+                                ...
```
Comment (jpwaves) [inline src/Kanbas/Courses/Quizzes/QuizList.tsx]: is this also something that you're still working on implementing, or is this being completed as part of someone else's work?

**Comment by mike10911** [comment]
Yep still working on styling and context menu! @jpwaves @alyssamui

File: src/Kanbas/Courses/Quizzes/QuizList.tsx (hunk 2 above)
Comment (mike10911) [reply src/Kanbas/Courses/Quizzes/QuizList.tsx]: ooo yep, good catch! That was from my old mock data I'll fix that right away!

File: src/Kanbas/Courses/Quizzes/Details/index.tsx (hunk 1 above)
Comment (mike10911) [reply src/Kanbas/Courses/Quizzes/Details/index.tsx]: Yeah this is part of my work! Working on the details editor as we speak!

**Review by alyssamui — APPROVED** [review]
lgtm

**Review by jpwaves — APPROVED** [review]

**Review by jessicasu77 — APPROVED** [review]
slayy

**Pull request merged**